dev_tools/tests/test_nyaml2nxdl.py

```diff
@@ -23,3 +25,35 @@ def test_conversion():
     assert root_content == new_root_content
     os.remove(yaml)
     os.remove(new_root)
+
+
+def test_doc():
+    """To test the doc style."""
+    pwd = Path(__file__).parent
+
+    doc_file = pwd / "data/doc_test.yaml"
+    ref_doc_file = pwd / "data/ref_doc_test.nxdl.xml"
+    out_doc_cile = pwd / "data/doc_test.nxdl.xml"  # doc_file.with_suffix('.nxdl.xml')
```

Comment:
This file is created in the data directory. It's better to use a temporary pytest directory (with, e.g., the [tmpdir or tmppath fixture](https://docs.pytest.org/en/6.2.x/tmpdir.html)) and write it there.

Reply:
That was something, I tried to use tmpdir on the first development of nyaml2nxdl, but by default the file will be generated in input file dir.

Reply:
Yes, to write you need to use the fixture to write it to a temporary directory. This is how it should be done. There should be no extra file created in the repository from the tests

Reply:
Ah I see you added the files to the repository. This should be removed and written to the temporary directory, otherwise the test would still report success even if the reader might fail in writing the file at all.

Reply:
addressed.